packages/ai-bot/helpers.ts

```diff
@@ -293,31 +295,55 @@ export function getRelevantCards(
   };
 }
 
-export function getTools(
+function getLastUserMessage(
   history: DiscreteMatrixEvent[],
   aiBotUserId: string,
-): Tool[] {
-  // TODO: there should be no default tools defined in the ai-bot, tools must be determined by the host
-  let searchTool = getSearchTool();
-  let tools = [searchTool as Tool];
-  // Just get the users messages
+): CardMessageEvent | undefined {
   const userMessages = history.filter((event) => event.sender !== aiBotUserId);
   // Get the last message
   if (userMessages.length === 0) {
-    // If the user has sent no messages, return tools that are available by default
-    return tools;
+    // If the user has sent no messages, return undefined
+    return undefined;
   }
   const lastMessage = userMessages[userMessages.length - 1];
   if (
     lastMessage.type === 'm.room.message' &&
-    lastMessage.content.msgtype === 'org.boxel.message' &&
-    lastMessage.content.data?.context?.tools?.length
+    lastMessage.content.msgtype === 'org.boxel.message'
   ) {
+    return lastMessage as CardMessageEvent;
+  }
+  return undefined;
+}
+
+export function getTools(
+  history: DiscreteMatrixEvent[],
+  aiBotUserId: string,
+): Tool[] {
+  const lastMessage = getLastUserMessage(history, aiBotUserId);
+  if (lastMessage && lastMessage.content.data?.context?.tools?.length) {
     return lastMessage.content.data.context.tools;
-  } else {
-    // If it's a different message type, or there are no tools, return tools that are available by default
-    return tools;
   }
+  return [];
+}
+
+export function getToolChoice(
+  history: DiscreteMatrixEvent[],
+  aiBotUserId: string,
+): ToolChoice {
+  let lastMessage = getLastUserMessage(history, aiBotUserId);
+  if (lastMessage && lastMessage.content.data?.context?.requireToolCall) {
+    let tools = lastMessage.content.data.context?.tools || [];
+    if (tools.length != 1) {
+      throw new Error('Forced tool calls only work with a single tool');
```

Comment:
Is this recoverable, or does it "break" a room forever?

Reply:
I'll test this before merging, with the check on the host we shouldn't get into this situation anyway but with the current setup this should be temporary as the list of commands and the requiredtoolcall is just on the last user message so the next valid one sent will be fine